components/applications-service/cmd/applications-publisher/main.go

```diff
@@ -126,6 +129,25 @@ func main() {
 		clientID = "applications-publisher-" + strconv.FormatInt(t.UnixNano(), 10)
 	}
 
+	// If the channel was specified, add the update config
+	if len(channel) > 0 {
```

Comment:
It's probably fine, but if no channel is provided, but an update-strategy, the latter will just be ignored; it won't raise an error.

Reply:
@srenatus not sure if you noticed but, we default the strategy to `at-once`, but the important thing here is that channel, we want to replicate the behavior of habitat and, if in Habitat you don't specify a channel, then the strategy won't be populated. TL DR no channel, no strategy.

Reply:
Yeah, what I imagined to possibly happen is this:
When I call `applications-publisher --channel dev --update-strategy foobar`, I get an error. When I run `applications-publisher --update-strategy foobar`, I *don't*. In a perfect world, specifying `--update-strategy` without specifying `--channel` would perhaps be an error.

However, I don't know if our cli arg lib makes this simple; if it doesn't, I doubt it's worth it 🤷‍♂

Reply:
I see, well I can add this check for sure. At the end, this is just a publisher tool to populate our DB but we want it to be as similar as a habitat supervisor as possible.

Reply:
Your call -- I have to little context to discern if this an issue. Probably not 😄